horovod/tensorflow/__init__.py

```diff
@@ -97,12 +98,18 @@ def allreduce(tensor, average=None, device_dense='', device_sparse='',
             if op == Adasum:
                 if ('CPU' not in tensor.device and has_gpu):
                     if nccl_built():
+                        if not check_num_rank_power_of_2(size() / local_size()):
```

Comment:
In this mode you also require that the GPUs per node are homogenous, right?  Otherwise the optimizers will have very different learning rates.  Perhaps you should add an assertion for that as well?

Reply:
Yes you are right, I forgot that case. will update it.